### PR TITLE
feat: defining schema for document with did signature

### PR DESCRIPTION
**Pull request by yehjxraymond**
Extends the `proof` structure to support documents that are directly signed with DID. 

- Convenient script to generated wrapped document from unwrapped ones (`/scripts/rewrap.ts`)
- Moved v2 specific types from common to v2 folder
- Hosted schema at schemata repo and updated references
- Updated schema at schemata to support `proofPurpose` and `revocation`
- Added sample structure for did signed document
- Added `proofPurpose` into all documents

### CODE REVIEW

**Review by Nebulis — APPROVED** [review]

File: src/2.0/types.ts [hunk 1]
```diff
@@ -1,7 +1,14 @@
 // types generated by quicktype during postinstall phase
 import { OpenAttestationDocument as OpenAttestationDocumentV2 } from "../__generated__/schema.2.0";
-import { SchemaId, ObfuscationMetadata, Proof } from "../shared/@types/document";
+import { SchemaId, ObfuscationMetadata, ProofPurpose, ProofType } from "../shared/@types/document";
 
+export interface Proof {
```
Comment (yehjxraymond) [inline src/2.0/types.ts:5]: Moving here since it's only used in v2 now

File: src/3.0/schema/batched-verifiable-credential-1.json [hunk 2]
```diff
@@ -1,11 +1,10 @@
 {
   "version": "https://schema.openattestation.com/3.0/schema.json",
-  "schema": "https://schema.openattestation.com/3.0/schema.json",
```
Comment (yehjxraymond) [inline src/3.0/schema/batched-verifiable-credential-1.json:3]: @Nebulis need to check with you if this is needed

File: tsconfig.json [hunk 3]
```diff
@@ -1,4 +1,7 @@
 {
+  "ts-node": {
```
Comment (yehjxraymond) [inline tsconfig.json:2]: Allows ts-node to detect modules. Used to run the rewrap.ts  script.

**Review by Nebulis — APPROVED** [review]

**Pull request merged**